evm/README.md

```diff
@@ -181,3 +181,72 @@ To get details of options provided in the command run:
+
+    - sourceChain: 'Axelarnet'
+    - sourceAddress: 'axelar10d07y265gmmuvt4z0w9aw880jnsr700j7v9daj'
+    - contractAddress = interchain governance address
+    - commandId = random bytes32 value
+
+2. Running the command above will output the raw batch data, the data hash, and the vald sign command that should be used to generate validator signatures. Here is an example output:
+
+    ```bash
+    axelard vald-sign evm-ethereum-2-3799882 [validator-addr] 0x2716d6d6c37ccfb1ea1db014b175ba23d0f522d7789f3676b7e5e64e4322731
+    ```
+
+    The validator address should be entered dynamically based on which validator is signing.
+
+3. Connect to AWS and switch to the correct context depending on which environment validators will be signing in.
+
+4. Get all the namespaces within this context:
+
+    ```bash
+    kubectl get ns
+    ```
+
+5. List pods for each validator namespace. Example for stagenet validator 1:
+
+    ```bash
+    kubectl get pods -n stagenet-validator1
+    ```
+
+6. Exec into the validator pod. Example for stagenet validator 1:
+
+    ```bash
+    kubectl exec -ti axelar-core-node-validator1-5b68db5c49-mf5q6 -n stagenet-validator1 -c vald -- sh
+    ```
+
+7. Within the validator container, run the vald command from step 2 with the correct validator address. Validator addresses can be found by appending this endpoint `/axelar/multisig/v1beta1/key?key_id=[KEY_ID]` to the Axelar lcd url in the chains config file. Running this command should result in a JSON output containing the signature, similar to the example below:
+
+    ```json
+    {
+        "key_id": "evm-ethereum-2-3799882",
+        "msg_hash": "9dd126c2fc8b8a29f92894eb8830e552b7caf1a3c25e89cc0d74b99bc1165039",
+        "pub_key": "030d0b5bff4bff1adb9ab47df8c70defe27bfae84b12b77fac64cdcc2263e8e51b",
+        "signature": "634cb96060d13c083572d5c57ec9f6441103532434824a046dec4e8dfa0645544ea155230656ee37329572e1c52df1c85e48f0ef5f283bc7e036b9fd51ade6661b",
+        "validator": "axelarvaloper1yfrz78wthyzykzf08h7z6pr0et0zlzf0dnehwx"
+    }
+    ```
+
+8. Create a signatures folder in the root directory of this repository. Repeat step 7 until enough signatures have been generated to surpass the threshold weight for the current epoch (threshold weight can be found at the same endpoint as validator addresses). Store each signature JSON in a separate file within the signatures folder, filenames do not have to follow any specific convention.
+
+9. Run the construct batch command passing in the raw batch data logged in step 1. This will generate the input bytes that can be passed directly into the execute command on Axelar gateway. You can also optionally pass the `--execute` command which will automatically execute the batch on the Axelar gateway contract.
+
+    ```bash
+    node evm/combine-signatures.js -e [ENV] -n [CHAIN_NAME] --action constructBatch --batchData [RAW_BATCH_DATA]
```

Comment:
This script is appending 0x in front of `pub_key` values of signature files, but the logs generated when signing within the validator already have 0x with it, so it would be good to handle this.

Reply:
@deanamiel ^

Reply:
Same with other values in the signature logged by validator

Reply:
When I generated the signatures they looked like the example below, so it was still necessary to append `0x`, do you see any signatures in different formats?
```
{
  "key_id": "evm-ethereum-2-3799882",
  "msg_hash": "9dd126c2fc8b8a...",
  "pub_key": "02884f2544...",
  "signature": "976da9bb8...",
  "validator": "axelarvaloper12yyp..."
}
```

Reply:
yep for me signatures look like this:

```json
{
  "key_id": "evm-fantom-3784501",
  "msg_hash": "0xd66d9...",
  "pub_key": "0x02407e...",
  "signature": "0xc072da18731dd498...",
  "validator": "axelarvaloper1n30vm3rym25kqrs3rpql..."
}
```

The command I used:
`axelard vald-sign evm-fantom-3784501 axelarvaloper1n30vm3rym25kqrs3rpql0m78wst5fgm
6zvr77h 0xd66d9988d2e618f2dc025bb7ae42b6e27a4f249e3275a6033026953c3fe3869c`

@milapsheth can diff. chains return a diff revert for axelard?

Reply:
Anyway @deanamiel we should be prepared for both formats and add `0x` only if it is not present.

Reply:
Weird, yeah I can update the logic to support both formats then